beer-garden/functionary
# #213 - Added file param support for the API

Closes #213 

## Introduced Changes
- The `/api/v1/tasks` endpoint now handles file input in the same way as the UI
  - The only caveat is that you cannot submit Tasks that rely on file parameters through the Swagger UI

## Testing
- Publish the demo package to Functionary
- Fetch an authentication token from the API
  - `curl -X POST http://localhost:8000/api/v1/api-token-auth -H 'accept: application/json' -F "username=" -F "password="`
- Submit a task to the `num_chars_api` test function
  - `curl -X POST http://localhost:8000/api/v1/tasks/ -H 'accept: application/json' -H 'X-Environment-Id: <env uuid>' -H 'Content-Type: multipart/form-data' -F "function=<function uuid>" -F "file.function=@/path/to/file" -F 'parameters={"other_param": "hello world"}' -H "Authorization: Token <your auth token>"`
- Verify task runs as expected
